flang/lib/Lower/OpenMP.cpp

```diff
@@ -110,6 +110,34 @@ static void gatherFuncAndVarSyms(
   }
 }
 
+static Fortran::lower::pft::Evaluation *
+getCollapsedEval(Fortran::lower::pft::Evaluation &eval, int collapseValue) {
+  // Return the Evaluation of the innermost collapsed loop, or the current
+  // evaluation, if there is nothing to collapse.
+  if (collapseValue == 0)
+    return &eval;
```

Comment:
Nit: Is it better to convert this to an assert (for > 0) and move this code to the parent function?

Reply:
I'll make this change in another commit.